rtl8821ce: bump to 5.5.2_34066.20200325

msfjarvis opened the pull request:
###### Motivation for this change

`linuxPackages_latest.rtl8821ce` failed to compile due to Linux ABI changes. The fixes were already made in the source repository but the package was not updated here, causing build failures as reported in #88068. With this change, I can build and use the `rtl8821ce` module successfully on the Linux 5.6.13 kernel.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] ~~Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))~~ No test exists
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

msfjarvis committed:
rtl8821ce: 5.2.5_1.26055.20180108 -> 5.5.2_34066.20200325

The rtl8821ce repository was updated to address for ABI changes to
Linux but our package was too far behind, resulting in breakages
as reported in #88068

Fixes: #88068

IvarWithoutBones reviewed (approved):
Working perfectly here :+1:

IvarWithoutBones commented:
after taking a second look at this:
Commits should be squased, with `rtl8821ce: 5.2.5_1.26055.20180108 -> 5.5.2_34066.20200325` as the title, according to [the contributing guidelines](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#submitting-changes)

msfjarvis commented:
> after taking a second look at this:
> diff looks good. though the commits should be squased, with `rtl8821ce: 5.2.5_1.26055.20180108 -> 5.5.2_34066.20200325` as the message, according to [the contributing guidelines](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#submitting-changes)

Thanks for pointing that out, not sure how I missed it. Rebased and squashed, should be good to go now :)

arte921 commented:
Any updates on this? I stil get compile errors when trying to install either `linuxPackages.rtl8821ce` or `linuxPackages_latest.rtl8821ce`, on default and latest kernels respectively.

msfjarvis commented:
> Any updates on this? I stil get compile errors when trying to install either `linuxPackages.rtl8821ce` or `linuxPackages_latest.rtl8821ce`, on default and latest kernels respectively.

@hhm0 is listed as the maintainer but he's been unresponsive so far and I'm not aware of who to notify or what the process is to either become the maintainer of this package or just have this update merged for the time being.

msfjarvis commented:
Almost forgot, while we await for the PR to be merged and backported to 20.03, you can fix this locally by doing the update in your configuration.nix file like this https://msfjarvis.dev/g/dotfiles/1e7e1306bace57267ef5085184c8fded7a1449ab

arte921 commented:
> Almost forgot, while we await for the PR to be merged and backported to 20.03, you can fix this locally by doing the update in your configuration.nix file like this https://msfjarvis.dev/g/dotfiles/1e7e1306bace57267ef5085184c8fded7a1449ab

Thanks! works perfectly :)

marsam commented:
Thank you! Sorry for the delay, please ping me if any PR gets stuck
backported to 20.03 in https://github.com/NixOS/nixpkgs/commit/0ba9fdc1e8e1817c7cf69b7fccb6fa5334114959
